### ISSUE TITLE
[android] App crashes when swapping a custom layer

### ISSUE THREAD

**Issue by Guardiola31337** [mapbox/mapbox-gl-native#8448]
Crash when pressing FAB in `CustomLayerActivity` using Samsung Galaxy S6 edge+:

![samsungsm-g928a](https://cloud.githubusercontent.com/assets/1668582/24043335/79944cb4-0b16-11e7-9d78-cb31a8ff1d8d.png)

**Platform:** Android
**Mapbox SDK version:** master branch

### Steps to trigger behavior

 1. Open `CustomLayerActivity`
 2. Press FAB to swap custom layer
 3. Crash

### Expected behavior
App doesn't crash

### Actual behavior
App crashes

![bugcustomlayer](https://cloud.githubusercontent.com/assets/1668582/24043456/ecf9b55e-0b16-11e7-8003-14941752dce6.gif)

**Comment by tobrun** [comment]
Is there a crash log availlable?

**Comment by Guardiola31337** [comment]
Nope, I was toying test app using [Samsung Remote Test Lab](http://developer.samsung.com/remotetestlab/rtlDeviceList.action) and it crashed.
BTW, I could get it because it's actually a physical device connected remotely.
Feel free to close, have to look further 😉

**Comment by DellScreen** [comment]
Attached a stack-trace obtained on an S8: [CustomLayerCrashOnS8.txt](https://github.com/mapbox/mapbox-gl-native/files/1074122/CustomLayerCrashOnS8.txt)

**Comment by tobrun** [comment]
From Samsung s6 edge, J3 and emulator:

```
java.lang.Error: vertexArray->bindVertexArray(value): Error GL_INVALID_OPERATION at /home/tvn/Mapbox/mapbox-gl-native/src/mbgl/gl/value.cpp:347 at com.mapbox.mapboxsdk.maps.NativeMapView.nativeRender(Native Method) at com.mapbox.mapboxsdk.maps.NativeMapView.render(NativeMapView.java:158) at com.mapbox.mapboxsdk.maps.MapView.onDraw(MapView.java:416) at android.view.View.draw(View.java:18318)
```

putting on milestone for visibility

**Comment by jfirebaugh** [comment]
Is this still reproducible?

**Comment by tobrun** [comment]
Was able reproduce this with a galaxy J3 on master (https://github.com/mapbox/mapbox-gl-native/commit/f053feb6d4aba1eaa44a35d050522b641c011b33)

> 07-23 23:45:54.545 2081-2081/com.mapbox.mapboxsdk.testapp E/AndroidRuntime: FATAL EXCEPTION: main
                                                                            Process: com.mapbox.mapboxsdk.testapp, PID: 2081
                                                                            java.lang.Error: vertexArray->bindVertexArray(value): Error GL_INVALID_OPERATION at /home/tvn/Mapbox/mapbox-gl-native/src/mbgl/gl/value.cpp:347
                                                                                at com.mapbox.mapboxsdk.maps.NativeMapView.nativeRender(Native Method)
                                                                                at com.mapbox.mapboxsdk.maps.NativeMapView.render(NativeMapView.java:125)
                                                                                at com.mapbox.mapboxsdk.maps.MapView.onDraw(MapView.java:425)
                                                                                at android.view.View.draw(View.java:16536)
...

**Comment by tobrun** [comment]
@jfirebaugh I'm thinking the source of the crash will be found in our example implementation [here](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/src/example_custom_layer.cpp). 
If I comment out the intialise/render code, it's not crashing anymore.

**Comment by ivovandongen** [comment]
@tobrun The crash on the Galaxy J3 could be due to bugs in the Adreno 300 series dealing with Vertex Arrays: https://github.com/mapbox/mapbox-gl-native/pull/10253

Not sure if that is also the case for the Samsung s6 edge(+). Should check that.

The crash log for the Samsung S8 seems different all together and should be investigated separately perhaps.

**Comment by zugaldia** [comment]
I was also able to reproduce this issue on a Nexus 5 emulator using `master` today, not on a Pixel XL device.

**Comment by ivovandongen** [comment]
I've added fixes in #11239. @Guardiola31337 @tobrun Would you please re-test with the devices you mentioned in this ticket. fwiw, I've reproduced the issue with a Galaxy Nexus.
